### PR TITLE
Natively support int truncation, don't guard on positive/negative

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122827

This doesn't entirely fix the original problem that prompted this, but
it seems to just be getting stuck in export constraint formatting now
which seems like progress to me.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by ezyang** [comment]
@avikchaudhuri the end to end still doesn't work because the export system first suffers from https://github.com/pytorch/pytorch/issues/121286 and when I poked it anyway it wants the user to tell it that after rounding the inputs are in bounds, or something like that.

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -1311,7 +1311,7 @@ def val_to_arg_str(self, s):
             import triton
 
         if isinstance(s, SymTypes):
-            return pexpr(sympy.expand(repr(s)))
+            return pexpr(s.node.expr)
```
Comment (ezyang) [inline torch/_inductor/codegen/wrapper.py:1314]: This is suspicious, in particular, I don't know why it was originally written this way. Testing this change alone in https://github.com/pytorch/pytorch/pull/122893

Comment (ezyang) [reply torch/_inductor/codegen/wrapper.py:1314]: cc @desertfire

**Review by avikchaudhuri — APPROVED** [review]
Thanks. Could you comment on the expected failure what `t` is when it fails the print assert? (`t` is the thing we try to split into `left`, `op`, `right`).

**Comment by ezyang** [comment]
@avikchaudhuri the value of t is `AssertionError: Round(0.5*height - 75.0) <= height`

**Pull request closed without merging**